testing/assert.ts

```diff
@@ -0,0 +1,179 @@
+// Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
+import { assertEqual as prettyAssertEqual } from "./pretty.ts";
+
+interface Constructor {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  new (...args: any[]): any;
+}
+
+/** Make an assertion, if not `true`, then throw. */
+export function assert(expr: boolean, msg = ""): void {
+  if (!expr) {
+    throw new Error(msg);
+  }
+}
+
+/**
+ * Make an assertion that `actual` and `expected` are equal, deeply. If not
+ * deeply equal, then throw.
+ */
+export function equal(actual: unknown, expected: unknown, msg?: string): void {
+  prettyAssertEqual(actual, expected, msg);
+}
+
+/**
+ * Make an assertion that `actual` and `expected` are strictly equal.  If
+ * not then throw.
+ */
+export function strictEqual(
+  actual: unknown,
+  expected: unknown,
+  msg?: string
+): void {
+  if (actual !== expected) {
+    let actualString: string;
+    let expectedString: string;
+    try {
+      actualString = String(actual);
+    } catch (e) {
+      actualString = "[Cannot display]";
+    }
+    try {
+      expectedString = String(expected);
+    } catch (e) {
+      expectedString = "[Cannot display]";
+    }
+    console.error(
+      "strictEqual failed. actual =",
+      actualString,
+      "expected =",
+      expectedString
+    );
+    if (!msg) {
+      msg = `actual: ${actualString} expected: ${expectedString}`;
+    }
+    throw new Error(msg);
+  }
+}
+
+/**
+ * Make an assertion that actual contains expected. If not
+ * then thrown.
+ */
+export function stringContains(
```

Comment:
Rename to `assertStringContains`

Reply:
Maybe `assertHasSubStr`

Reply:
Maybe `assertContains`

Reply:
strContains seems more legitimate because i wanted to add arrContains assert after.